### DISCUSSION TITLE
Hot reloading

### DISCUSSION THREAD

**Discussion by halturin** [ergo-services/ergo#105] [Polls]
One of the coolest features in the BEAM VM is hot reloading. Not sure how often is used, but it definitely the coolest one. What if we create something similar in the Ergo Framework? Golang provides 'plugin' out of the box. There are a lot of controversial opinions about this feature in terms of stability and limitations. Please share your experience with it. How it was? Painfully? Usefully?

**Comment by hdahlheim** [comment]
In the time that I have used elixir (about 2-3 years), I never used hot code reloading. :) It's a cool feature, but the as far as I can tell not many people are using it (at least in the elixir community).  It also does not fit a containerized deployment workflows. 

I agree it would be a cool, but it's not a must-have for me ;)

**Reply by sleipnir** [reply]
> In the time that I have used elixir (about 2-3 years), I never used hot code reloading. :) It's a cool feature, but the as far as I can tell not many people are using it (at least in the elixir community). It also does not fit a containerized deployment workflows.
> 
> I agree it would be a cool, but it's not a must-have for me ;)

I agree. This is definitely not a differential nowadays

**Comment by leonlee2013** [comment]
The hot reloading is very useful, and we use it a lot, and is a feature that Erlang has over Go. It would be great if it could be implemented in Ergo and would greatly improve Ergo's competitiveness.

**Comment by Northward712** [comment]
If you want to add plugin, you will not be able to use this feature under windows
In addition, overloading Plugin will not cause the old one to be uninstalled, it will still exist in memory (maybe), so it may lead to memory growth.

**Reply by halturin** [reply]
There is another approach https://github.com/eh-steve/goloader#comparison-with-plugin
But I'm still on the fence.

**Comment by samuelmanzanera** [comment]
I used a lot hot reload for my Elixir app (blockchain node). It fact it was one of the main reason to rewrite our app in Elixir (coming from Go).
Previously I used Go plugins but this was a bit painful in terms of pointers, function signatures,...
In that sense Erlang hot reload is fantastic and works out of the box for most scenarios. 

But nowadays to implement it in Golang maybe I would use technologies like WebAssembly this could be a smart move in terms of adoption and easiness of use.
